ui/app/adapters/secrets-engine-path.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+/**
+ * General use adapter to access specified paths on secrets engines
+ * For example /:backend/config is a typical use case for this adapter
+ */
+
+import ApplicationAdapter from 'vault/adapters/application';
+import { encodePath } from 'vault/utils/path-encoding-helpers';
+
+export default class KubernetesConfigAdapter extends ApplicationAdapter {
+  namespace = 'v1';
+
+  // define path value in extending class or pass into method directly
+  getURL(backend, path) {
```

Comment:
Maybe I'm confused on when we should prefix methods with the underscore. My understanding was that they were for private methods. You use it for `_saveRecord` but then should it also be used for `_getUrl`?

Reply:
Yep good call. To keep it consistent I will add the underscore to `getURL`.